src/main/kotlin/org/rust/lang/core/completion/RsKeywordCompletionContributor.kt

```diff
@@ -142,7 +142,12 @@ class RsKeywordCompletionContributor : CompletionContributor(), DumbAware {
 
     private fun baseDeclarationPattern(): PsiElementPattern.Capture<PsiElement> =
         psiElement()
-            .withParent(or(psiElement<RsPath>(), psiElement<RsModItem>(), psiElement<RsFile>()))
+            .withParent(or(
+                psiElement<RsPath>(),
+                psiElement<RsModItem>(),
+                psiElement<RsMembers>().withParent(psiElement<RsTraitOrImpl>()),
```

Comment:
I think we shouldn't try to change `baseDeclarationPattern` because it covers all necessary cases for top-level declarations.
We need a separate completion provider for `trait` and `impl` blocks because only several items can be placed inside them. For example, we can't create `enum` inside `trait` block.
The current implementation can suggest invalid keywords
![image](https://user-images.githubusercontent.com/2539310/62973412-ad760500-be1e-11e9-887b-36ce54e86332.png)